homeassistant/components/growatt_server/sensor.py

```diff
@@ -579,91 +711,76 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
+    entities = [
+        GrowattInverter(
+            probe,
+            name=f"{name} Total",
+            unique_id=f"{plant_id}-{description.key}",
+            description=description,
         )
+        for description in TOTAL_SENSOR_TYPES
+    ]
 
     # Add sensors for each device in the specified plant.
     for device in devices:
         probe = GrowattData(
             api, username, password, device["deviceSn"], device["deviceType"]
         )
-        sensors = []
+        sensor_descriptions = ()
         if device["deviceType"] == "inverter":
-            sensors = INVERTER_SENSOR_TYPES
+            sensor_descriptions = INVERTER_SENSOR_TYPES
         elif device["deviceType"] == "storage":
             probe.plant_id = plant_id
-            sensors = STORAGE_SENSOR_TYPES
+            sensor_descriptions = STORAGE_SENSOR_TYPES
         elif device["deviceType"] == "mix":
             probe.plant_id = plant_id
-            sensors = MIX_SENSOR_TYPES
+            sensor_descriptions = MIX_SENSOR_TYPES
         else:
             _LOGGER.debug(
                 "Device type %s was found but is not supported right now",
                 device["deviceType"],
             )
 
-        for sensor in sensors:
-            entities.append(
+        entities.extend(
+            [
                 GrowattInverter(
                     probe,
-                    f"{device['deviceAilas']}",
-                    sensor,
-                    f"{device['deviceSn']}-{sensor}",
+                    name=f"{device['deviceAilas']}",
+                    unique_id=f"{device['deviceSn']}-{description.key}",
+                    description=description,
                 )
-            )
+                for description in sensor_descriptions
+            ]
+        )
 
     async_add_entities(entities, True)
 
 
 class GrowattInverter(SensorEntity):
     """Representation of a Growatt Sensor."""
 
-    def __init__(self, probe, name, sensor, unique_id):
+    entity_description: GrowattSensorEntityDescription
+
+    def __init__(
+        self, probe, name, unique_id, description: GrowattSensorEntityDescription
+    ):
         """Initialize a PVOutput sensor."""
-        self.sensor = sensor
         self.probe = probe
-        self._name = name
-        self._state = None
-        self._unique_id = unique_id
+        self.entity_description = description
 
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{self._name} {SENSOR_TYPES[self.sensor][0]}"
-
-    @property
-    def unique_id(self):
-        """Return the unique id of the sensor."""
-        return self._unique_id
-
-    @property
-    def icon(self):
-        """Return the icon of the sensor."""
-        return "mdi:solar-power"
+        self._attr_name = f"{name} {description.name}"
+        self._attr_unique_id = unique_id
+        self._attr_icon = "mdi:solar-power"
```

Comment:
This is static and can be set as a class attribute on the `GrowattInverter` class